File: VbFileSearcher/AccessorAdapter.vb
Imports System.IO.MemoryMappedFiles

''' <summary>
'''     A class which creates an IReadable out of a MemoryMappedViewAccessor
''' </summary>
Public Class AccessorAdapter
    Implements IReadable

    ReadOnly _accessor As MemoryMappedViewAccessor

    Public Sub New(accessor As MemoryMappedViewAccessor)
        _accessor = accessor
    End Sub


    Public Function CanRead() As Boolean Implements IReadable.CanRead
        Return _accessor.CanRead
    End Function


    Public Function Capacity() As Long Implements IReadable.Capacity
        Return _accessor.Capacity
    End Function

    Public Function ReadUInt16(offset As Long) As UShort Implements IReadable.ReadUInt16
        Return _accessor.ReadUInt16(offset)
    End Function
End Class

File: FileSearcherProject/AccessorAdapter.cs
using System.IO.MemoryMappedFiles;

namespace FileSearcherProject
{
  public class AccessorAdapter : IReadable
  {
    #region Private Members

    private readonly MemoryMappedViewAccessor _accessor;

    #endregion

    #region Constructor(s)

    public AccessorAdapter(MemoryMappedViewAccessor accessor)
    {
      _accessor = accessor;
    }

    #endregion

    #region Public Properties

    /// <summary>
    /// returns whether or not file can be read
    /// </summary>
    public bool CanRead { get { return _accessor.CanRead; } }

    /// <summary>
    /// returns capacity of file
    /// </summary>
    public long Capacity { get { return _accessor.Capacity; } }

    #endregion

    #region Public Methods

    public ushort ReadUInt16(long offset)
    {
      return _accessor.ReadUInt16(offset);
    }

    #endregion

  }
}
